src/aggregate/aggregate_request.c

```diff
@@ -126,7 +126,13 @@ static int parseRequiredFields(AREQ *req, ArgsCursor *ac, QueryError *status){
   int requiredFieldNum = AC_NumArgs(&args);
   const char** requiredFields = array_new(const char*, requiredFieldNum);
   for(size_t i=0; i < requiredFieldNum; i++) {
-    requiredFields = array_append(requiredFields, AC_GetStringNC(&args, NULL));
+    const char *s = AC_GetStringNC(&args, NULL); {
+      if(!s) {
+        array_free(requiredFields);
```

Comment:
Don't you need to also free the strings that was already added to `requiredFields`?

Reply:
Also do we have test for this case?

Reply:
it is `const char *`, and a direct result of `AC_GetString` which is either `RedisModule_StringPtrLen` or `AC_CURRENT`. Anyway, it is not a copy

Reply:
If its not a copy then how do you know it will not be freed when you try to use it on `req->requiredFields`. Also what about a test that check this flow?